conn.go

```diff
@@ -852,7 +849,10 @@ func (c *conn) startTLS() stateFunc {
 		if c.connectTimeout != 0 {
 			_ = tlsConn.SetWriteDeadline(time.Now().Add(c.connectTimeout))
 		}
-		c.err = tlsConn.Handshake()
+		if err := tlsConn.Handshake(); err != nil {
+			done <- err
+			// TODO: return?
```

Comment:
I assume the answer is 'no', primarily because the 'tlsConn' is still active, even if handshaking fails?

Who's responsible for closing 'c.net' below, basically?

Reply:
Also, given how the code is structured you could probably just `done <- tlsConn.Handshake()` unconditionally.

Reply:
It seems weird to me that we'd swap out `c.net` and set `c.tlsComplete` to `true` if this failed, which is why I added the comment.  However, since this is just a refactor, I didn't want to change any existing behavior.